### PR TITLE
refactor mempool CheckTx and broadcast

### PR DESCRIPTION
**Pull request by xzav3r**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: app/ante/ante.go [hunk 1]
```diff
@@ -33,6 +31,14 @@ func NewAnteHandler(ak auth.AccountKeeper, evmKeeper EVMKeeper, sk types.SupplyK
 	return func(
 		ctx sdk.Context, tx sdk.Tx, sim bool,
 	) (newCtx sdk.Context, err error) {
+		if ctx.IsWrappedCheckTx() {
+			if _, ok := tx.(evmtypes.MsgEthereumTx); ok {
+				// check nonce for MsgEthereumTx
+				return sdk.ChainAnteDecorators(NewNonceVerificationDecorator(ak), NewIncrementSenderSequenceDecorator(ak))(ctx, tx, sim)
```
Comment (g302ge) [inline app/ante/ante.go]: In the NonceVerificationDecorator should use the From which calculate from the EthSigVerification. So if need add a NewEthSigVerificationDecorator for this

Comment (g302ge) [reply app/ante/ante.go]: StdTx also needs the Sequence Ante to check the account sequence if is valid maybe add this 

```go
if _, ok := tx.(auth.StdTx); ok {
        return sdk.ChainAnteDecorators(authante.NewIncrementSequenceDecorator(ak))
}
```

Comment (xzav3r) [reply app/ante/ante.go]: The confident node has done the check.
So, it is not necessary to redo this for the receiver node if the confident node is fully trusted.

Comment (g302ge) [reply app/ante/ante.go]: But, in this deoractor https://github.com/okex/exchain/blob/6af20879d4a935cbf45318f4ea549c08db5f2b65/app/ante/eth.go#L246

you can found that the inner check have to use the signature to recover from and use it to do next

Comment (xzav3r) [reply app/ante/ante.go]: done

Comment (g302ge) [reply app/ante/ante.go]: LGTM

File: libs/tendermint/mempool/reactor.go [hunk 2]
```diff
@@ -106,9 +114,14 @@ func newMempoolIDs() *mempoolIDs {
 // NewReactor returns a new Reactor with the given config and mempool.
 func NewReactor(config *cfg.MempoolConfig, mempool *CListMempool) *Reactor {
 	memR := &Reactor{
-		config:  config,
-		mempool: mempool,
-		ids:     newMempoolIDs(),
+		config:           config,
+		mempool:          mempool,
+		ids:              newMempoolIDs(),
+		nodeKeyWhitelist: make(map[string]struct{}),
```
Comment (g302ge) [inline libs/tendermint/mempool/reactor.go:120]: Maybe in the future we could use the Value directly. better for []byte or string ?

**Review by g302ge — DISMISSED** [review]
LGTM

File: app/ante/newante.go [hunk 3]
```diff
@@ -1,124 +1,123 @@
...
-	default:
-		return ctx, sdkerrors.Wrapf(sdkerrors.ErrUnknownRequest, "invalid payload transaction type: %T", payloadTx)
-	}
-	newCtx, err = payloadAnteHandler(newCtx, payloadTx, sim)
-	logger.Info("Payload tx anteHandler", "err", err)
-
-	return
-}
-
+//import (
```
Comment (xiangjianmeng) [inline app/ante/newante.go:3]: It's suggested to delete these codes

**Review by xiangjianmeng — DISMISSED** [review]

**Review by MinZhu09 — APPROVED** [review]

**Review by KamiD — APPROVED** [review]

**Pull request merged**